osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -510,12 +510,27 @@ public bool SelectBeatmap(BeatmapInfo? beatmapInfo, bool bypassFilters = true)
             if (beatmapInfo?.Hidden != false)
                 return false;
 
-            foreach (CarouselBeatmapSet set in beatmapSets)
+            foreach (var carouselItem in root.Items)
```

Comment:
I dunno about this. You're moving away from using properties that clean code up in just one place, but leaving said properties used elsewhere. This just feels like a wrong direction.

Reply:
I've reverted `BeatmapCarousel` changes altogether. Should've just went for obvious top 3 but instead stuck in a rabbit hole.
Can agree that saving 5K objects here out of 2M isn't worth code readability.

Reply:
Put another way, these are gen0 allocs. They get freed almost immediately. And they aren't even happening every frame, but only once per beatmap selection. It's a "best case" allocation, and fixing will likely only result in a minor performance improvement and little-to-no GC pressure reduction.

In addition, what I was trying to get at is that if this is something we want to change, it should be done at a higher level (ie. other usages should not be using `beatmapSets` still). *If* that's something we want to do.

When looking for allocations you want to reduce, you may benefit in filtering the profiler to show things that reach gen1 or gen2 (using snapshot comparison mode, not alloc mode), as those are the most expensive (and cause the larger stutters) in smaller numbers.